dbuf-lsp/src/action_handler/rename.rs

```diff
@@ -22,124 +22,143 @@ pub fn renameable_symbol(symbol: &Symbol) -> bool {
             dependency: _,
         } => true,
         Symbol::Field {
+            t: _,
             constructor: _,
             field: _,
         } => true,
         Symbol::Alias {
             t: _,
             branch_id: _,
-            name: _,
+            alias: _,
+        } => false,
+        Symbol::Constructor {
+            t: _,
+            constructor: _,
         } => false,
-        Symbol::Constructor(_) => false,
         Symbol::None => false,
     }
 }
 
 /// Check if symbol can be renamed to new_name without conflicts.
-pub fn renameable_to_symbol(symbol: &Symbol, new_name: &String, ast: &ElaboratedAst) -> Result<()> {
+pub fn renameable_to_symbol(symbol: &Symbol, new_name: &str, ast: &ElaboratedAst) -> Result<()> {
     if new_name.is_empty() {
-        return rename_errors::rename_to_empty_error();
+        return RenameError::ToEmpty.into();
     }
     if dbuf_language::get_builtin_types().contains(new_name) {
-        return rename_errors::rename_to_builtin_type_error();
+        return RenameError::ToBuiltin.into();
     }
     if dbuf_language::get_keywords().contains(new_name) {
-        return rename_errors::rename_to_keyword_error();
+        return RenameError::ToKeyword.into();
     }
 
     match symbol {
         Symbol::Type(t) => {
-            if dbuf_language::get_builtin_types().contains(t) {
-                return rename_errors::rename_of_builtin_type_error();
-            }
-            if !dbuf_language::is_correct_type_name(new_name) {
-                return rename_errors::rename_to_bad_type_error(new_name);
-            }
             if t == new_name {
-                return rename_errors::rename_to_old_error();
+                return RenameError::ToPrevious.into();
```

Comment:
Some kind of a `test` function, like described [here](https://www.reddit.com/r/rust/comments/xw5lk4/function_like_assert_that_returns_a_result_instead/), can be used to simplify control flow here a bit